pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -585,6 +585,13 @@ private boolean processStreamEvents(MessageBatch messagesAndOffsets, long idlePi
             realtimeRowsConsumedMeter =
                 _serverMetrics.addMeteredTableValue(_metricKeyName, ServerMeter.REALTIME_ROWS_CONSUMED, 1,
                     realtimeRowsConsumedMeter);
+            long lastMileIngestionDelayMs = System.currentTimeMillis() - msgMetadata.getRecordIngestionTimeMs();
+            if (lastMileIngestionDelayMs >= 0) {
+              _segmentLogger.warn("ADDING REALTIME DATA FRESHNESS: delta="
```

Comment:
I think this is frequently going to have a non-zero lag, so I'm not sure there is any value in logging it. At the very least, consider logging at info level, since it is not indicative of an operational issue.

Also, consider using consistent format for the logger, like capitalization and {} usage:
```
_segmentLogger.warn("Ingested realtime partition with a lag of {}ms", lastMileIngestionDelayMs)
```

Reply:
If logging is required here, log it if delay is above a certain time period say 15-30 minutes. This will otherwise fill up the logs, also if we could add a log limiter, which won't print this if delay is already printed in the past x minutes.

Reply:
Logging is not needed. metric is good enough. we can set alerts on the metric